sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/fluentcore/dag/TaskGroup.java

```diff
@@ -399,7 +400,15 @@ public void addPostRunDependentTaskGroup(TaskGroup dependentTaskGroup) {
             }
             readyTaskEntry = super.getNext();
         }
-        return Flux.mergeDelayError(32, observables.toArray(new Flux[0]));
+        return Flux.mergeOrdered(32,
```

Comment:
There is some difference on "DelayError" part as well. Might need to check a bit on `processFaultedTaskAsync` method.

Reply:
mergeOrdered will automatically delay error to the last, due to it cannot compare item with error

Reply:
Added the mergeOrdered test about it.

Reply:
If convenient, try write a few tests for it (positive cases and negative cases)?